### PR TITLE
ipn: delete domainsForProxying, require explicit DNS search domains (mapver 9)

### PR DESCRIPTION
**Pull request by bradfitz**
Previously the client had heuristics to calculate which DNS search domains
to set, based on the peers' names. Unfortunately that prevented us from
doing some things we wanted to do server-side related to node sharing.

So, bump MapRequest.Version to 9 to signal that the client only uses the
explicitly configured DNS search domains and doesn't augment it with its own
list.

Updates tailscale/corp#1026

### CODE REVIEW

**Comment by bradfitz** [comment]
I won't merge this until the control side is deployed, otherwise it'll break MagicDNS for people running at HEAD.

**Review by josharian — APPROVED** [review]
I don't have full context for this, so feel free to wait for a review from someone else, but from what I know, looks good.

**Review by crawshaw — APPROVED** [review]

**Comment by bradfitz** [comment]
The server-side parts are deployed now so I was able to test this and it doesn't quite work yet.

Will finish it up this week.

**Review by danderson — APPROVED** [review]

**Pull request merged**
